DAOS-9721 test: update tests to use get_container

daltonbohning opened the pull request:
Test-tag: pr container TestWithTelemetryIOLatency test_ec_offline_rebuild_single test_ec_degrade_single_value test_ec_aggregation_default SimpleCreateDeleteTest DaosContainterSecurityTest CreateContainterACLTest DeleteContainerACLTest GetContainerACLTest OverwriteContainerACLTest UpdateContainerACLTest
Skip-unit-tests: true
Skip-fault-injection-test: true

- Update tests to use get_container instead of TestContainer directly
- Misc avocado tag cleanup
- Misc code cleanup

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [x] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [x] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on EL 8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-11769/5/testReport/(root)/

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-11769/7/execution/node/1038/log

daltonbohning commented:
Existing failures in:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11769/7/tests/
- `scrubber/target_auto_eviction.py` - https://daosio.atlassian.net/browse/DAOS-12857

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-11769/8/execution/node/1036/log

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-11769/8/execution/node/1037/log

daltonbohning commented:
Failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-11769/8/tests/
- `daos_test/suite.py:DaosCoreTest.test_daos_rebuild_ec`
  - Seen in daily run https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/master/8297/tests
  - Assume ticket will be created for daily failure
- `scrubber/target_auto_eviction.py` - https://daosio.atlassian.net/browse/DAOS-12857
- `pool/create_capacity.py` - https://daosio.atlassian.net/browse/DAOS-12899

daltonbohning commented on src/tests/ftest/nvme/pool_capacity.py:
```diff
@@ -115,8 +113,10 @@ def test_create_delete(self, num_pool=2, num_cont=5, total_count=100):
                 nvme_size_begin[val] = self.pool[-1].get_pool_free_space("NVME")
-                for cont_val in range(0, num_cont):
-                    cont[cont_val] = TestContainer(self.pool[-1])
+                # To be fixed by DAOS-12974
+                for _ in range(num_cont):
+                    # self.get_container(self.pool[-1])
+                    pass
```
https://daosio.atlassian.net/browse/DAOS-12974

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

shimizukko commented on src/tests/ftest/nvme/pool_capacity.py:
```diff
@@ -101,7 +100,6 @@ def test_create_delete(self, num_pool=2, num_cont=5, total_count=100):
                 None
```
```suggestion
```
I guess we don't need this. Same for `run_test()`

daltonbohning commented:
Tests fixed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11769/9/testReport/
- DAOS-12978
  - FIXED - security/cont_acl.py
- DAOS-12987
  - FIXED - erasurecode/aggregation.py:EcodAggregationOff
  - NOT-RAN - erasurecode/offline_rebuild.py:EcodOfflineRebuild
  - FIXED erasurecode/offline_rebuild_aggregation.py:EcodAggregationOffRebuild 
  - FIXED erasurecode/rebuild_disabled.py:EcodDisabledRebuild
  - FIXED erasurecode/restart.py:EcodServerRestart

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
